### PR TITLE
Unify data model to rely on splash response

### LINKED ISSUE #84: Provide clear Input structure for `/extract_meta` endpoint
Currently, the input data model looks as follows:

```python

class Input(BaseModel):
    url: str = Field(..., description="The base url of the scraped website.")
    html: Optional[str] = Field(
        default=None,
        description="Everything scraped from the website as text.",
    )
    headers: Optional[str] = Field(
        default=None,
        description="The response header originally received together with the content.",
    )
    har: Optional[str] = Field(
        default=None, description="The har object interpretable as json."
    )
    allow_list: Optional[ListTags] = Field(
        default=ListTags(),
        description="A list of key:bool pairs. "
        "Any metadata key == True will be extracted. "
        "If this list is not given, all values will be extracted.",
    )
    debug: Optional[bool] = Field(
        default=True,
        description="Developer flag to receive more information through API",
    )
    bypass_cache: Optional[bool] = Field(
        default=False,
        description="Bypass the cache (true) or not in evaluation.",
    )

```

From this it is not clear what is supposed to happen for the different combinations of provided values - E.g. what should happen if `har` and `html` are provided? Should we use the html block from the `har` or the `html` value?

To resolve this, I propose to change the input model to either accept only the URL, or URL plus `har`. If the `har`is provided, the extraction would not issue a request to Splash to generate it's own har.

I.e. I propose to drop the html and header fields, as they will either be taken from the `har` that is delivered as part of the input, or from the response issued to splash if no `har` was provided.

### PR DESCRIPTION
**Pull request by MRuecklCC**
Resolves #84

### CODE REVIEW

**Review by MRuecklCC — COMMENTED** [review]
Some minor things that still need to be addressed!

File: src/app/models.py [hunk 1]
```diff
@@ -229,17 +231,17 @@ class Config:
...
+        ..., description="The URL where the content was crawled from."
     )
-    headers: Optional[str] = Field(
+    splash_response: Optional[SplashResponse] = Field(
         default=None,
-        description="The response header originally received together with the content.",
-    )
-    har: Optional[str] = Field(
-        default=None, description="The har object interpretable as json."
+        # fixme: Specify expected HAR structure e.g. with JSON Schema from
```
[the remark this answers is not in the document]
Comment (RMeissnerCC) [reply src/app/models.py:239]: So why do you not remove it?

File: src/app/splash_models.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+
+class Cookie(BaseModel):
+    """See http://www.softwareishard.com/blog/har-12-spec/#cookies"""
+
+    name: str
+    value: str
+    httpOnly: bool
+    secure: bool
+    # currently not used and hence not required
```
Comment (RMeissnerCC) [inline src/app/splash_models.py:20]: Then why do you have this commented? I would remove it, if it is not necessary

File: src/app/splash_models.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    The response received from a splash render-html request.
+
+    Not all fields received in the response are listed here for now. However, what is listed here is required.
+    """
+
+    url: HttpUrl
+    html: str
+    har: HAR
+
+    # class Config:
```
[the remark this answers is not in the document]
Comment (RMeissnerCC) [reply src/app/splash_models.py:71]: You mean, you want a test that hands over a dict with more keys than the expected fields? I.e., to check that ignore=True by default? Sounds good :)

[the remark this answers is not in the document]
Comment (RMeissnerCC) [reply src/app/splash_models.py:71]: Ticket please ;) Or add it to one of the already existing issues

File: tests/splash-response-google.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
...
+  "html": "<html></html>",
```
Comment (RMeissnerCC) [inline tests/splash-response-google.json:11]: Why this change?

**Review by RMeissnerCC — CHANGES_REQUESTED** [review]
Nice changes. I like the models :)

File: src/app/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MRuecklCC) [reply src/app/models.py:239]: I will! I sometimes review my own PRs to find things like that and comment on them to not forget :D

File: src/app/splash_models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MRuecklCC) [reply src/app/splash_models.py:71]: Ill add it to this PR. It belongs here with the introduction of all the model classes.

File: src/app/splash_models.py (hunk 2 above)
Comment (MRuecklCC) [reply src/app/splash_models.py:20]: will do!

File: tests/splash-response-google.json (hunk 4 above)
Comment (MRuecklCC) [reply tests/splash-response-google.json:11]: oh, that slipped in accidentally! need to revert that!

File: src/app/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MRuecklCC) [reply src/app/models.py:239]: done

File: src/app/splash_models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MRuecklCC) [reply src/app/splash_models.py:71]: done

File: src/app/splash_models.py (hunk 2 above)
Comment (MRuecklCC) [reply src/app/splash_models.py:20]: done

File: tests/splash-response-google.json (hunk 4 above)
Comment (MRuecklCC) [reply tests/splash-response-google.json:11]: done

**Review by RMeissnerCC — APPROVED** [review]
Looks good. Please merge :)
